mne/coreg.py

```diff
@@ -1245,3 +1248,632 @@ def _scale_xfm(subject_to, xfm_fname, mri_name, subject_from, scale,
+
+        self._extra_points_filter = None
+        self._dig = _get_data_as_dict_from_dig(
+            dig=self._info['dig'],
+            exclude_ref_channel=False
+        )
+        # adjustments
+        self._dig['rpa'] = np.array([self._dig['rpa']])
+        self._dig['nasion'] = np.array([self._dig['nasion']])
+        self._dig['lpa'] = np.array([self._dig['lpa']])
+
+        self._setup_mri(subjects_dir=self._subjects_dir,
+                        subject=self._subject)
+        self.reset()
+        self._update_params(
+            rot=self._default_parameters[:3],
+            tra=self._default_parameters[3:6],
+            sca=self._default_parameters[6:9],
+        )
+
+    def _setup_mri(self, subjects_dir, subject):
+        # find high-res head model (if possible)
+        high_res_path = _find_head_bem(subject, subjects_dir, high_res=True)
+        low_res_path = _find_head_bem(subject, subjects_dir, high_res=False)
+        if high_res_path is None and low_res_path is None:
+            raise RuntimeError("No standard head model was "
+                               f"found for subject {subject}")
+        if high_res_path is not None:
+            self.bem_high_res = self._read_surface(high_res_path)
+        else:
+            self.bem_high_res = self._read_surface(low_res_path)
+        if low_res_path is None:
+            # This should be very rare!
+            warn('No low-resolution head found, decimating high resolution '
+                 'mesh (%d vertices): %s' % (len(self.bem_high_res.surf.rr),
+                                             high_res_path,))
+            # Create one from the high res one, which we know we have
+            rr, tris = decimate_surface(self.bem_high_res.surf.rr,
+                                        self.bem_high_res.surf.tris,
+                                        n_triangles=5120)
+            # directly set the attributes of bem_low_res
+            self.bem_low_res = complete_surface_info(dict(rr=rr, tris=tris),
+                                                     copy=False, verbose=False)
+        else:
+            self.bem_low_res = self._read_surface(low_res_path)
+
+        # Set MNI points
+        try:
+            fids = get_mni_fiducials(subject, subjects_dir)
+        except Exception:  # some problem, leave at origin
+            self.mni_points = None
+        else:
+            self.mni_points = np.array([f['r'] for f in fids], float)
+
+        # find fiducials file
+        fid_files = _find_fiducials_files(subject, subjects_dir)
+        if len(fid_files) > 0:
+            fid_filename = fid_files[0].format(subjects_dir=subjects_dir,
+                                               subject=subject)
+        else:
+            fid_filename = None
+        if fid_filename is None or not op.exists(fid_filename):
+            self.fid_points = self.mni_points
+        else:
+            self.fid_points = _fiducial_coords(*read_fiducials(fid_filename))
```

Comment:
I would not keep mni_points as attribute unless it’s necessary

Also I would put this logic of guessing the fiducials in a dedicated private function. Function that would only be called if ‘auto' mode is used